api/v1alpha1/openapi.yaml

```diff
@@ -2539,10 +2539,42 @@ components:
         - type: object
           properties:
             inline:
-              type: object
-              additionalProperties: true
+              type: array
+              items:
+                $ref: '#/components/schemas/FileSpec'
           required:
           - inline
+    FileSpec:
+      type: object
+      properties:
+        path:
+          type: string
+          description: The absolute path to the file on the device. Note that any existing file will be overwritten.
+        contents:
+          type: string
+          description: The URL-encoded or base64-encoded contents of the file.
+        contentsEncoding:
+          type: string
+          description: How the contents are encoded.
+          enum:
+          - base64
+          - url
```

Comment:
I assume this is still inspired by Ignition, so the `url` would refer to the ability to pull in content from external sources via "http, https, tftp, s3, arn, gs" schemes. Is that the case and, if so, who would pull and would this conflict with the config providers?

Wondering whether we should limit ourselves to base64.

Reply:
No, it just specifies how the contents are encoded (either base64 encoding or URL encoding).  All of our examples currently use URL encoding so I assumed we wanted to support both.

Reply:
Ah, the data URI scheme `data:content/type;base64,` that Ignition prefixes, right? Do we need the ability to express MIME type, character encoding, etc? Could we autodetect that? Default that?

Reply:
"Contents" now is just the data, without the data URL format.  Then "ContentsEncoding" tells us how the data is encoded so that it can later be properly decoded.